apps/vx-scan/backend/src/app.ts

```diff
@@ -35,14 +36,117 @@ const debug = rootDebug.extend('app');
 
 type NoParams = never;
 
+/**
+ * An interface for interacting with USB drives. We inject this into the app so
+ * that we can easily mock it in tests.
+ */
+export interface Usb {
+  getUsbDrives: () => Promise<UsbDrive[]>;
+}
+
+/**
+ * Possible errors that can occur during configuration (currently there's only one).
+ */
+export type ConfigurationError = 'no_ballot_package_on_usb_drive';
+
 function buildApi(
   machine: PrecinctScannerStateMachine,
   interpreter: PrecinctScannerInterpreter,
   workspace: Workspace,
+  usb: Usb,
   logger: Logger
 ) {
   const { store } = workspace;
+
   return grout.createApi({
+    async configureFromBallotPackageOnUsbDrive(): Promise<
+      Result<void, ConfigurationError>
+    > {
+      assert(!store.getElectionDefinition());
+      const [usbDrive] = await usb.getUsbDrives();
+      assert(usbDrive?.mountPoint !== undefined, 'No USB drive mounted');
```

Comment:
This could just be a normal error, right?

```suggestion
      if (!usbDrive?.mountPoint) {
        return err('no_usb_drive_mounted');
      }
```

Reply:
^ I did notice that this error will be surfaced by Grout as a 500 instead of a properly handled error result object, but figured that was fine (and perhaps even intentional) since we only ever call this endpoint once a USB is mounted, so this error is never expected. That said, I could imagine a race condition where the USB is mounted and then quickly removed, triggering this case on the server

Reply:
It was intentional to signify "this should never happen." I agree there's a possible race condition, but I wasn't sure if it was possible/worth handling. In general I imagine the same issue applies during the actual reading from the USB - if it's removed, things will explode. So that makes me think there's not a huge advantage to catching the USB removed right before L67 vs right after L67.

This does raise another question for me: how we can gracefully handle the USB being removed during the configuration process? Looking at the old code I removed, there was a catch-all handler that showed an "Unknown error" message. I'm planning to implement something like that for the whole app once I convert the frontend to react-query.